packages/cli/src/databases/repositories/sharedWorkflow.repository.ts

```diff
@@ -125,4 +125,20 @@ export class SharedWorkflowRepository extends Repository<SharedWorkflow> {
 
 		return transaction.save(newSharedWorkflows);
 	}
+
+	async findWithFields(candidateIds: string[], { fields }: { fields: string[] }) {
+		return this.find({
+			where: {
+				workflowId: In(candidateIds),
+			},
+			select: fields as FindOptionsSelect<SharedWorkflow>,
```

Comment:
Thanks! I thought of this but the call sites [didn't typecheck](https://share.cleanshot.com/fhRVFzV9) and apparently this is because `FindOptionsSelectByString<E>` was [deprecated](https://github.com/typeorm/typeorm/blob/c22e30f1d2ae0c197d17ff8bd9bb65c0a2014542/src/find-options/FindOptionsSelect.ts#L43), so removing this assertion for `FindOptionsSelect<E>` would require more refactoring. I didn't spend longer as the focus of the PR is moving typeorm to the persistence layer. Definitely more work remains to be done there once we finish here.